stable_baselines/common/env_checker.py

```diff
@@ -0,0 +1,142 @@
+                      "We recommend you to flatten the observation "
+                      "to have only a 1D vector")
+
+    if warn and isinstance(action_space, spaces.Box):
+        # if the if-block is not separated `action_space.low` will be evaluated even though
+        # `isinstance(action_space, spaces.Box)` is False...
+        if np.abs(action_space.low) != np.abs(action_space.high) or np.abs(action_space.low) > 1 or np.abs(action_space.high) > 1:
+            warnings.warn("We recommend you to use a symmetric and normalized Box action space (range=[-1, 1]) "
+                          "cf https://stable-baselines.readthedocs.io/en/master/guide/rl_tips.html")
+
+
+    # ============ Check the returned values ===============
+
+    # because env inherits from gym.Env, we assume that `reset()` and `step()` methods exists
+    obs = env.reset()
+
+    if not isinstance(observation_space, spaces.Tuple):
+        assert not isinstance(obs, tuple), "The `reset()` method should only return one value, not a tuple"
+
+    # The check for a GoalEnv is done by the base class
+    if not is_goal_env:
+        if isinstance(observation_space, spaces.Discrete):
+            assert isinstance(obs, int), "The observation returned by `step()` method must be an int"
+        elif enforce_array_obs:
+            assert isinstance(obs, np.ndarray), "The observation returned by `reset()` method must be a numpy array"
+
+    assert observation_space.contains(obs), "The observation returned by the `reset()` method does not match the given observation space"
+
+    # Sample a random action
+    action = action_space.sample()
+    data = env.step(action)
+
+    assert len(data) == 4, "The `step()` method must return four values: obs, reward, done, info"
+
+    # Unpack
+    obs, reward, done, info = data
+
+    if not is_goal_env:
+        if isinstance(observation_space, spaces.Discrete):
+            assert isinstance(obs, int), "The observation returned by `step()` method must be an int"
+        elif enforce_array_obs:
+            assert isinstance(obs, np.ndarray), "The observation returned by `step()` method must be a numpy array"
+    assert observation_space.contains(obs), "The observation returned by the `step()` method does not match the given observation space"
+
+    # We also allow int because the reward will be cast to float
+    assert isinstance(reward, (float, int)), "The reward returned by `step()` must be a float"
+    assert isinstance(done, bool) or float(done) in (0, 1), "The `done` signal must be a boolean"
+    assert isinstance(info, dict), "The `info` returned by `step()` must be a python dictionary"
+
+    if isinstance(env, gym.GoalEnv):
+        # For a GoalEnv, the keys are check at reset
+        assert reward == env.compute_reward(obs['achieved_goal'], obs['desired_goal'], info)
+
+    # TODO: should we check metadata['render.modes'] ?
```

Comment:
should we? :)

Reply:
I'm not sure about it (I think I will only emit a warning if `render.modes` is not present and check if the `render.modes` correspond really to what can be used)